src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -336,19 +351,21 @@ export class CellModel implements ICellModel {
 
 					// requestExecute expects a string for the code parameter
 					content = Array.isArray(content) ? content.join('') : content;
-					const future = kernel.requestExecute({
-						code: content,
-						stop_on_error: true
-					}, false);
-					this.setFuture(future as FutureInternal);
-					this.fireExecutionStateChanged();
-					// For now, await future completion. Later we should just track and handle cancellation based on model notifications
-					let result: nb.IExecuteReplyMsg = <nb.IExecuteReplyMsg><any>await future.done;
-					if (result && result.content) {
-						this.executionCount = result.content.execution_count;
-						if (result.content.status !== 'ok') {
-							// TODO track error state
-							return false;
+					if (!this.checkForAdsCommandMagic()) {
```

Comment:
So if the cell has a magic anywhere in it you won't execute any of the rest of the cell? Is that normal notebook behavior for magics?

Reply:
So "real" magics are built into kernels. Since we want this functionality to work for all kernels, we need to bypass the kernel execution mechanism in this particular case, and define it in a central location (i.e. here 😄 )

Reply:
You could execute the magic(s) and then remove them from the script before sending to Jupyter, right? Pretty hacky I guess - does Jupyter not have the way to contribute magics that are outside the scope of specific kernels?